MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2005,7 +2005,26 @@ public string GetVideoQualityParam(EncodingJobInfo state, string videoEncoder, E
             }
 
             var profile = state.GetRequestedProfiles(targetVideoCodec).FirstOrDefault() ?? string.Empty;
-            profile = WhiteSpaceRegex().Replace(profile, string.Empty);
+            profile = WhiteSpaceRegex().Replace(profile, string.Empty).ToLowerInvariant();
+
+            if (string.Equals("h264", targetVideoCodec, StringComparison.OrdinalIgnoreCase)
+                || string.Equals("avc", targetVideoCodec, StringComparison.OrdinalIgnoreCase))
+            {
+                profile = _videoProfilesH264.Contains(profile, StringComparison.OrdinalIgnoreCase) ? profile : string.Empty;
+            }
+            else if (string.Equals("h265", targetVideoCodec, StringComparison.OrdinalIgnoreCase)
```

Comment:
I believe it will never equal `h265` because we assign `hevc` ourselves: https://github.com/jellyfin/jellyfin/blob/a1c7dd68f9bb9c79d969795e866acd664c029ab5/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs#L2004

Possible rewrite (pseudo-code):
```cs
var videoProfiles;

if (h264) videoProfiles = _videoProfilesH264;
else if (hevc) videoProfiles = _videoProfilesH265;
else if (av1) videoProfiles = _videoProfilesAv1;
else videoProfiles = new string[]; // or static dummy array

if (!videoProfiles.Contains(profile, StringComparison.OrdinalIgnoreCase)
{
    profile = string.Empty;
}
```

Reply:
Done. IMHO `avc` is not needed too.